tests/test_package_creation.py

```diff
@@ -0,0 +1,101 @@
+import pytest
+import pytest_copie
+import subprocess
+
+def assert_successful_project_creation(result):
+    """Basic assertions that indicate the copier was able to create a project"""
+    assert result.exit_code == 0
+    assert result.exception is None
+    assert result.project_dir.is_dir()
+
+
+def assert_directory_structure(result, package_name = "example_package"):
+    """Test to ensure that the default directory structure ws created correctly"""
+    assert (result.project_dir / f"src/{package_name}").is_dir()
+    assert (result.project_dir / f"tests/{package_name}").is_dir()
+
+
+def assert_black_runs_successfully(result):
+    """Test to ensure that the black linter runs successfully on the project"""
+    # run black with `--check` to look for lint errors, but don't fix them.
+    black_results = subprocess.run(
+        ["python", "-m", "black", "--check", (result.project_dir / "src")],
+        cwd=result.project_dir
+    )
+
+    assert black_results.returncode == 0
+
+
+def assert_pylint_runs_successfully(result):
+    """Test to ensure that the pylint linter runs successfully on the project"""
+    # run pylint to ensure that the hydrated files are linted correctly
+    pylint_results = subprocess.run(
+        ["python", "-m", "pylint",
+         "--recursive=y",
+         "--rcfile=./src/.pylintrc",
+         (result.project_dir / "src")],
+        cwd=result.project_dir
+    )
+
+    assert pylint_results.returncode == 0
+
+def test_all_defaults(copie):
+    """Test that the default values are used when no arguments are given.
+    Ensure that the project is created and that the basic files exist.
+    """
+
+    # run copier to hydrate a temporary project
+    result = copie.copy()
+
+    assert_successful_project_creation(result)
+
+    assert_directory_structure(result)
+
+    assert_pylint_runs_successfully(result)
```

Comment:
This is silly. But. Within the pytest-copie invocation, within the hydrated project. can we then run `pytest` to make sure that the example code will build and tests will pass?

Reply:
I believe that would be possible! Good call out :)

Reply:
Ok, after attempting to implement something, I quickly ran into the issue that the hydrated project needs to be pip installed into an environment in order for unit tests to run. 

I'm spending a moment looking into scratch environments that might help here. This might be a difficult thing :(